mod/display.php

```diff
@@ -55,10 +55,25 @@ function display_init(&$a) {
 		}
 		if (count($r)) {
 			if ($r[0]["id"] != $r[0]["parent"])
-				$r = q("SELECT `id`, `author-name`, `author-link`, `author-avatar`, `network`, `body`, `uid` FROM `item`
+				$r = q("SELECT `id`, `author-name`, `author-link`, `author-avatar`, `network`, `body`, `uid`, `owner-link` FROM `item`
 					WHERE `item`.`visible` AND NOT `item`.`deleted` AND NOT `item`.`moderated`
 						AND `id` = %d", $r[0]["parent"]);
 
+			if (($itemuid != local_user()) AND local_user()) {
+				// Do we know this contact but we haven't got this item?
+				// Copy it to our local storage.
```

Comment:
can you extend the comment why this is needed. e.g `so the user would be able to comment on old posts`?